pkg/sql/pgwire/conn.go

```diff
@@ -40,12 +41,28 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/log"
 	"github.com/cockroachdb/cockroach/pkg/util/mon"
 	"github.com/cockroachdb/cockroach/pkg/util/ring"
+	"github.com/cockroachdb/cockroach/pkg/util/syncutil"
 	"github.com/cockroachdb/cockroach/pkg/util/timeutil"
 	"github.com/cockroachdb/errors"
 	"github.com/cockroachdb/redact"
 	"github.com/lib/pq/oid"
 )
 
+// atomicWriter is a wrapper around an io.Writer which presents a thread-safe interface to it.
+type atomicWriter struct {
```

Comment:
for the record i dont think this is a good idea. but the pgwire.conn interface is very loosey goosey and i dont know of a better way to allow concurrent calls to it, which is now required due to the asynchronous notifications. open to suggestions!